lib/String.fz

```diff
@@ -574,6 +574,87 @@ public String ref : property.equatable, property.hashable, property.orderable is
     String.type.from_bytes (utf8.as_list.reverse.drop_while c->c.is_ascii_white_space).reverse
 
 
+  # pad this string at the end with spaces such that its `codepoint_length` is at least `n`.
+  #
+  public pad(n i32)
+  pre
+    debug: n >= 0
+  post
+    debug: result.codepoint_length >= n
+    debug: ((result.codepoint_length != n): result = String.this)
+  =>
+    pad " " n
+
+
+  # pad this string at the beginning with spaces such that its `codepoint_length` is at least `n`.
+  #
+  public pad_left(n i32)
+  pre
+    debug: n >= 0
+  post
+    debug: result.codepoint_length >= n
+    debug: ((result.codepoint_length != n): result = String.this)
+  =>
+    pad_left " " n
+
+
+  # pad this string at the beginning and at the end with spaces such that its `codepoint_length` is at least `n`.
+  #
+  public pad_center(n i32)
+  pre
+    debug: n >= 0
+  post
+    debug: result.codepoint_length >= n
+    debug: ((result.codepoint_length != n): result = String.this)
+  =>
+    pad_center " " n
+
+
+  # pad this string at the end with `p` such that its `codepoint_length` is at least `n`.
+  #
+  public pad(p String, n i32)
+  pre
+    safety: p.codepoint_length > 0
+    debug: n >= 0
+  post
+    debug: result.codepoint_length >= n
+    debug: ((result.codepoint_length >= n+p.codepoint_length): result = String.this)
+  =>
+    pl := p.codepoint_length
```

Comment:
Cool.
I think readability might improve a bit if expression `(max 0 (n-codepoint_length+pl-1)/pl)` was pulled out and given a proper name.